### PR TITLE
[NCL-3886]: Relax milestone version restrictions to support Continuous Delivery - 1.4

### PR DESCRIPTION
**Pull request by jbartece**

### CODE REVIEW

**Comment by jbartece** [comment]
retest this please

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/ProductMilestoneRest.java [hunk 1]
```diff
@@ -41,6 +42,7 @@
     private Integer id;
 
     @NotNull(groups = {WhenCreatingNew.class, WhenUpdating.class})
+    @Pattern(regexp = "^[0-9]+\\.[0-9]+(\\.[0-9]+)?\\.[\\w]+$", message = "Version doesn't match the required pattern ^[0-9]+\\.[0-9]+(\\.[0-9]+)?\\.[\\w]+$")
```
Comment (matejonnet) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/ProductMilestoneRest.java]: Can you describe in a comment what this regex does as they are hard to read.

File: model/src/main/java/org/jboss/pnc/model/Artifact.java [hunk 2]
```diff
@@ -17,23 +17,7 @@
...
-import javax.persistence.Index;
-import javax.persistence.JoinColumn;
-import javax.persistence.ManyToMany;
-import javax.persistence.ManyToOne;
-import javax.persistence.PersistenceException;
-import javax.persistence.PreRemove;
-import javax.persistence.SequenceGenerator;
-import javax.persistence.Table;
-import javax.persistence.UniqueConstraint;
+import javax.persistence.*;
```
[the remark this answers is not in the document]
Comment (jbartece) [reply model/src/main/java/org/jboss/pnc/model/Artifact.java]: OK, will reconfigure IDE

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/ProductMilestoneRest.java (hunk 1 above)
Comment (jbartece) [reply rest-model/src/main/java/org/jboss/pnc/rest/restmodel/ProductMilestoneRest.java]: OK

**Comment by jbartece** [comment]
@matejonnet Could you review the PR, please?

**Pull request merged**